server/src/test/java/io/crate/analyze/InsertAnalyzerTest.java

```diff
@@ -484,15 +498,22 @@ public void test_insert_from_values_with_mixed_type_used_target_column_type() th
     @Test
     public void test_pk_col_generated_from_col_with_default() throws Exception {
         AnalyzedInsertStatement stmt = e.analyze("insert into doc.pk_generated_default(col1) values (1)");
-        Asserts.assertThat(stmt.subQueryRelation().outputs())
+        assertThat(stmt.subQueryRelation().outputs())
             .satisfiesExactly(
                 isReference("col1", DataTypes.INTEGER));
-        Asserts.assertThat(stmt.primaryKeySymbols()).satisfiesExactly(
+        assertThat(stmt.primaryKeySymbols()).satisfiesExactly(
             isInputColumn(0),
             isFunction("subtract",
                        isFunction("multiply",
                                   isFunction("add", isLiteral(10), isLiteral(20)),
                                   isInputColumn(0)),
                        isLiteral(2)));
     }
+
+    @Test
+    public void test_cannot_insert_nulls_into_float_vector() {
+        assertThatThrownBy(() -> e.analyze("insert into doc.ai(f) values([1.1, null, 2.2])"))
```

Comment:
I'd tend to create a test case in `FloatVectorTypeTest` against implicitCast directly.
Has less overhead and might be easier to find if looking for existing test cases of the type.

Reply:
I've changed, my thought of analyzer test, was to show that the exception is forwarded and doesn't get converted into a cast exception (because of the implicit cast in place).